### ISSUE TITLE
aws.vpc_get_all_subnets filters does not work.

### ISSUE THREAD

**Issue by xydinesh** [StackStorm-Exchange/stackstorm-aws#11]
I can not filter subnets by VPC as now.

`st2 run aws.vpc_get_all_subnets filters="{'vpc_id':'vpc-5640903f'}"`

>.
id: 587710463aae7b55c673548c
status: failed
parameters: 
  filters: '{''vpc_id'':''vpc-5640903f''}'
result: 
  exit_code: 1
  result: None
  stderr: "Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 242, in <module>
    obj.run()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 154, in run
    output = action.run(**self._parameters)
  File "/opt/stackstorm/packs/aws/actions/run.py", line 20, in run
    return self.do_method(module_path, cls, action, **kwargs)
  File "/opt/stackstorm/packs/aws/actions/lib/action.py", line 129, in do_method
    resultset = getattr(obj, action)(**kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/boto/vpc/__init__.py", line 1150, in get_all_subnets
    self.build_filter_params(params, filters)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/boto/ec2/connection.py", line 126, in build_filter_params
    filters = dict(filters)
ValueError: dictionary update sequence element #0 has length 1; 2 is required
"
  stdout: ''

**Comment by xydinesh** [comment]
It is possible that my filter syntax is wrong, but I have tried different ways. I think, boto is receiving a string into dict() which cause the error.

**Comment by userlocalhost** [comment]
Good catch!

I could reproduce it in my environment.
```
(vagrant@ubuntu2:~$ st2 run aws.vpc_get_all_subnets filters='vpc_id=vpc-8e5aaceb'
..
id: 588091f288fa3f2422acc45e
status: failed
parameters: 
  filters: vpc_id=vpc-8e5aaceb
result: 
  exit_code: 1
  result: None
  stderr: "Traceback (most recent call last):
  File "/home/vagrant/st2/st2common/st2common/runners/python_action_wrapper.py", line 249, in <module>
    obj.run()
  File "/home/vagrant/st2/st2common/st2common/runners/python_action_wrapper.py", line 154, in run
    output = action.run(**self._parameters)
  File "/opt/stackstorm/packs/aws/actions/run.py", line 20, in run
    return self.do_method(module_path, cls, action, **kwargs)
  File "/opt/stackstorm/packs/aws/actions/lib/action.py", line 129, in do_method
    resultset = getattr(obj, action)(**kwargs)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/boto/vpc/__init__.py", line 1150, in get_all_subnets
    self.build_filter_params(params, filters)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/boto/ec2/connection.py", line 126, in build_filter_params
    filters = dict(filters)
ValueError: dictionary update sequence element #0 has length 1; 2 is required
"
  stdout: ''
vagrant@ubuntu2:~$ 
```

> I have tried different ways. I think, boto is receiving a string into dict() which cause the error.

Yes. I also think the reason is the incorrect type of 'filters' parameter.

So, I'll correct it!

**Comment by emedvedev** [comment]
Fixed in #13.
